src/js/facility-locator/components/Listbox.jsx

```diff
@@ -0,0 +1,161 @@
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import { truncate, kebabCase } from 'lodash';
+import { benefitsServices, facilityTypes, vetCenterServices } from '../config';
+import { getServices } from '../utils/helpers.js';
+
+class Listbox extends Component {
+
+  isSelectedOption = (option, type) => {
+    return this.props.currentQuery[type] === option;
+  }
+
+  renderSelectOptions = () => {
+    const { currentQuery: { facilityType, serviceType } } = this.props;
+    const services = getServices(facilityType, benefitsServices, vetCenterServices);
+    this.props.resetOptions();
+    return (
+      <ul
+        className="dropdown"
+        role={!this.props.isDisabled ? 'listbox' : undefined}
+        id="service-list">
+        {services && services.map((k, i) => {
+          const defaultSelected = !serviceType && k === 'All';
+          const isSelected = this.isSelectedOption(k, 'serviceType');
+          const isHovered = defaultSelected || isSelected;
+          const serviceOptionClasses = classNames({
+            'dropdown-option': true,
+            'facility-option': true,
+            'is-hovered': isHovered
+          });
+
+          return (
+            <li
+              tabIndex={this.props.dropdownActive ? '0' : '-1'}
+              role="option"
+              aria-selected={isHovered}
+              className={serviceOptionClasses}
+              key={k}
+              value={k}
+              ref={ elem => { this.props.options[i] = elem; }}
```

Comment:
I think React discourages modifying props passed into components. Is there another way we can handle this? I'm also not quite understanding what the options array is used for.

Reply:
Good point, I will pass a callback down to update this. The options array stores the refs for the options and for the dropdown button to help manage the focus.